internal/drivers/manager.go

```diff
@@ -38,6 +38,9 @@ type (
 	poolEntry struct {
 		sync.Mutex
 		Pool
+
+		LockByInstance map[string]*sync.Mutex
+		InstLock       sync.Mutex
```

Comment:
why do we have both a lock by instance map and an InstLock; and another mutex?

Reply:
InstLock is to ensure that fetching instance lock happens over a lock